src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -82,11 +91,53 @@ object WdlExpression {
     val parseTree = parser.parse_e(tokens, new WdlSyntaxErrorFormatter(terminalMap))
     new WdlExpression(parseTree.toAst)
   }
+
+  def toString(ast: AstNode, highlighter: SyntaxHighlighter = NullSyntaxHighlighter): String = {
```

Comment:
Yeah I was only suggesting that we should have some way of dynamically configuring Cromwell and that the Reader Monad was one potential option for this.  That was just thinking out loud and definitely not meant as a gating factor for this PR.  It may take us a while to pick a configuration approach we can feel good about.

I'm glad you went through the video.  Supplementing my half-attentive viewing of this with some blog posts, I also came away worrying about God objects and what seemed like consistently trivial examples.   i.e. will it scale? :smile: